web/src/lib/components/album-page/albums-list.svelte

```diff
@@ -0,0 +1,287 @@
+  import { mdiDeleteOutline } from '@mdi/js';
+  import { orderBy } from 'lodash-es';
+  import { onMount } from 'svelte';
+  import { flip } from 'svelte/animate';
+  import MenuOption from '$lib/components/shared-components/context-menu/menu-option.svelte';
+  import ContextMenu from '$lib/components/shared-components/context-menu/context-menu.svelte';
+  import AlbumsTable from '$lib/components/album-page/albums-table.svelte';
+  import { handleError } from '$lib/utils/handle-error';
+  import { searchAlbum } from './albums-controls.svelte';
+
+  export let albums: AlbumResponseDto[];
+
+  let shouldShowEditAlbumForm = false;
+  let selectedAlbum: AlbumResponseDto;
+  let albumToDelete: AlbumResponseDto | null;
+  let contextMenuPosition: OnShowContextMenuDetail = { x: 0, y: 0 };
+  let contextMenuTargetAlbum: AlbumResponseDto | undefined = undefined;
+
+  $: {
+    for (const key of sortByOptions) {
+      if (key.title === $albumViewSettings.sortBy) {
+        albums = key.sortFn(key.sortDesc, albums);
+        $albumViewSettings.sortDesc = key.sortDesc; // "Save" sortDesc
+        $albumViewSettings.sortBy = key.title;
+        break;
+      }
+    }
+  }
+  $: isShowContextMenu = !!contextMenuTargetAlbum;
+  $: albumsFiltered = albums.filter((album) => album.albumName.toLowerCase().includes($searchAlbum.toLowerCase()));
+
+  onMount(async () => {
+    await removeAlbumsIfEmpty();
+  });
+
+  function showAlbumContextMenu(contextMenuDetail: OnShowContextMenuDetail, album: AlbumResponseDto): void {
+    contextMenuTargetAlbum = album;
+    contextMenuPosition = {
+      x: contextMenuDetail.x,
+      y: contextMenuDetail.y,
+    };
+  }
+
+  function closeAlbumContextMenu() {
+    contextMenuTargetAlbum = undefined;
+  }
+
+  async function handleDeleteAlbum(albumToDelete: AlbumResponseDto): Promise<void> {
+    await deleteAlbum({ id: albumToDelete.id });
+    albums = albums.filter(({ id }) => id !== albumToDelete.id);
+  }
+
+  const chooseAlbumToDelete = (album: AlbumResponseDto) => {
+    contextMenuTargetAlbum = album;
+    setAlbumToDelete();
+  };
+
+  const setAlbumToDelete = () => {
+    albumToDelete = contextMenuTargetAlbum ?? null;
+    closeAlbumContextMenu();
+  };
+
+  const handleEdit = (album: AlbumResponseDto) => {
+    selectedAlbum = { ...album };
+    shouldShowEditAlbumForm = true;
+  };
+
+  const deleteSelectedAlbum = async () => {
+    if (!albumToDelete) {
+      return;
+    }
+    try {
+      await handleDeleteAlbum(albumToDelete);
+    } catch {
+      notificationController.show({
+        message: 'Error deleting album',
+        type: NotificationType.Error,
+      });
+    } finally {
+      albumToDelete = null;
+    }
+  };
+
+  const removeAlbumsIfEmpty = async () => {
+    try {
+      for (const album of albums) {
+        if (album.assetCount == 0 && album.albumName == '') {
+          await handleDeleteAlbum(album);
+        }
+      }
+    } catch (error) {
+      console.log(error);
+    }
```

Comment:
To keep the try block small (and because it makes more sense to me semantically) I would move the for and if out of the try block